src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/BasePostsCreator.java

```diff
@@ -0,0 +1,69 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import com.wikia.webdriver.elements.mercury.components.discussions.desktop.CategoryPills;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public abstract class BasePostsCreator extends BasePageObject implements PostsCreator {
+
+  private final CategoryPills categoryPills;
+
+  protected abstract String getBaseCssClassName();
+
+  protected abstract WebElement getPostsCreator();
+
+  protected abstract WebElement getSignInDialog();
+
+  protected abstract WebElement getGuidelinesMessageCloseButton();
+
+  protected abstract WebElement getDescriptionTextarea();
+
+  protected abstract WebElement getAddCategoryButton();
+
+  protected abstract WebElement getSubmitButton();
+
+  public BasePostsCreator() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)